cmd/libsnap-confine-private/snap.c

```diff
@@ -52,10 +62,37 @@ bool sc_security_tag_validate(const char *security_tag, const char *snap_name)
 	if (status != 0 || matches[1].rm_so < 0) {
 		return false;
 	}
+	// if we expect a component name (a non-null string was passed in here),
+	// then we need to make sure that the regex captured a component name
+	if (component_name != NULL) {
+		// don't allow empty component names, only allow NULL as an indication
+		// that we don't expect a component name.
+		if (strlen(component_name) == 0) {
+			return false;
+		}
+		// fail if the security tag doesn't contain a component name and we
+		// expected one
+		if (matches[7].rm_so < 0) {
+			return false;
+		}
+
+		size_t component_name_len = strlen(component_name);
```

Comment:
This matches what we do below for `snap_name`, and we cannot use `len+1` since the matched string won't have a null terminator that we can compare to.

Reply:
Right, I did not notice that one of the strings was not zero-terminated.